### PR TITLE
fix: sanitize repo creds in error messages (#12309)

### LINKED ISSUE #12309: Github App Private Key printed on auth failure
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ ] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

When trying to create an application, I saw in both a popup in the frontend UI, *and* in the logs from the server:
```
time="2023-02-07T03:07:17Z" level=info msg="finished unary call with code InvalidArgument" error="rpc error: code = InvalidArgument desc = application spec for argocd-admin is invalid: InvalidSpecError: repository not accessible: repositories not accessible: &Repository{Repo:https://github.com/james-callahan/gitops,Username:,Password:,SSHPrivateKey:,ConnectionState:ConnectionState{Status:,Message:,ModifiedAt:<nil>,},InsecureIgnoreHostKey:false,Insecure:false,EnableLFS:false,TLSClientCertData:,TLSClientCertKey:,Type:,Name:,InheritedCreds:true,EnableOCI:false,GithubAppPrivateKey:-----BEGIN RSA PRIVATE KEY-----\nTHEACTUALPRIVATEKEY\n-----END RSA PRIVATE KEY-----\n,GithubAppId:12345,GithubAppInstallationId:12345,GitHubAppEnterpriseBaseURL:,Proxy:,Project:,GCPServiceAccountKey:,ForceHttpBasicAuth:false,}" grpc.code=InvalidArgument grpc.method=Update grpc.service=application.ApplicationService grpc.start_time="2023-02-07T03:07:17Z" grpc.time_ms=38.558 span.kind=server system=grpc
```
where `THEACTUALPRIVATEKEY` is the actual private key for my github app.

**Expected behavior**

secret fields shouldn't be printed as part of error messages

**Version**

```shell
argocd-server: v2.6.0+acc554f
```

### PR DESCRIPTION
**Pull request by crenshaw-dev**
Fixes https://github.com/argoproj/argo-cd/issues/12309

### CODE REVIEW

**Review by wanghong230 — APPROVED** [review]

**Review by blakepettersson — APPROVED** [review]

**Review by jannfis — APPROVED** [review]
LGTM

File: pkg/apis/application/v1alpha1/repository_types.go [hunk 1]
```diff
@@ -265,6 +266,13 @@ func (m *Repository) CopySettingsFrom(source *Repository) {
 	}
 }
 
+func (m *Repository) StringForLogging() string {
+	if m == nil {
+		return ""
+	}
+	return fmt.Sprintf("&Repository{Repo: %q, Type: %q, Name: %q, Project: %q}", m.Repo, m.Type, m.Name, m.Project)
+}
```
Comment (jannfis) [inline pkg/apis/application/v1alpha1/repository_types.go:275]: Hm, we could make this a true implementation of the `Stringer` interface instead?

Comment (crenshaw-dev) [reply pkg/apis/application/v1alpha1/repository_types.go:275]: `.String()` is already implemented by the protobuf-generated code for the `Repository` struct. That's the method which is producing the unsanitized string.

Comment (jannfis) [reply pkg/apis/application/v1alpha1/repository_types.go:275]: Yeah, that makes sense. Thanks!

**Comment by crenshaw-dev** [comment]
Cherry-picked onto release-2.6 for 2.6.1.

**Pull request merged**